msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,215 @@
+
+1. [Register their single page application](#register-your-single-page-application)
+1. [Add trusted brokers](#add-trusted-brokers)
+1. [Initialize public client application](#initialize-public-client-application)
+1. [Call an API](#call-an-api)
+
+### Register your single-page application
+
+You need to create a Microsoft Azure App registration for your add-in on the Azure portal. The app registration must have a name, supported account type, and SPA redirect. If your add-in requires additional app registration beyond NAA and SSO, see [Register your single-page application.](/entra/identity-platform/scenario-spa-app-registration)
+
+### Add trusted brokers
+
+To enable nested authentication, your application must actively configure a redirect URI to indicate the Microsoft identity platform that your application allows itself to be brokered by supported hosts. The redirect URI of the application must be of type **Single Page Application** and conform to the following scheme:
+
+```
+brk-<broker_application_id>://<your_domain>
+```
+
+Where <broker_application_id> is the alias of the broker or brokers you wish to trust and <your_domain> is the fully qualified domain name where your application is hosted. For example, **brk-multihub://contoso.com**. If your app has been upgraded to also run in Outlook and Microsoft365.com (in addition to Teams) then you just need to add one Redirect URI:
+
+```http
+brk-multihub://<your_domain>
+```
+
+Your domain should only include the origin and not sub-paths. For example:
+
+* Correct: brk-multihub://myapp.teams.microsoft.com
+* Incorrect: brk-multihub://myapp.teams.microsoft.com/go
+
+### Initialize public client application
+
+> [!NOTE]
+> Ensure that you initialize TeamsJS before you initialize MSAL (Microsoft Authentication Library), else Nested app authentication might fail.
+
+You need to initialize MSAL and get an instance of the public client application. This is used to get access tokens when needed.
+
+```javascript
+import {
+  AccountInfo,
+  IPublicClientApplication,
+  createNestablePublicClientApplication,
+} from "@azure/msal-browser";
+
+const msalConfig = {
+  auth: {
+    clientId: "your_client_id",
+    authority: "https://login.microsoftonline.com/{your_tenant_id}",
+  },
+};
+
+let pca: IPublicClientApplication;
+
+export function initializePublicClient() {
+  console.log("Starting initializePublicClient");
+  return createNestablePublicClientApplication(msalConfig).then(
+    (result) => {
+      console.log("Client app created");
+      pca = result;
+      return pca;
+    }
+  );
+}
+```
+
+### Acquire your first token
+
+The tokens acquired by MSAL.js through nested app authentication are issued for your Azure app registration ID. The MSAL.js handles token acquisition for user authentication. It tries to get an access token silently, and if that fails, it prompts the user interactively.  The token is then used to call the Microsoft Graph API.
```

Comment:
>The token is then used to call the Microsoft Graph API.

Seems like this could leave somebody with the impression that this _only_ works for Graph... do we want to say "_or other Entra ID protected resource_" or something to this effect?

Reply:
updated